File: bench/cases/generic-pair-1m/main.go
package main

import "fmt"

type Pair[A any, B any] struct {
	Fst A
	Snd B
}

func loopSum(n int64) int64 {
	var sum int64 = 0
	for i := int64(0); i < n; i++ {
		p := &Pair[int64, int64]{Fst: i, Snd: i + 1}
		sum = sum + p.Fst + p.Snd
	}
	return sum
}

func main() {
	fmt.Println(loopSum(1000000))
}

File: bench/cases/generic-pair-1m/main.ts
type Pair<A, B> = { fst: A; snd: B }

function loopSum(n: number): number {
  let sum = 0
  for (let i = 0; i < n; i = i + 1) {
    const p: Pair<number, number> = { fst: i, snd: i + 1 }
    sum = sum + p.fst + p.snd
  }
  return sum
}

console.log(loopSum(1000000))
